// md2/datepicker/calendar.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, ElementRef, Input, Output, EventEmitter } from '@angular/core';
export var Md2Clock = (function () {
    function Md2Clock(_element) {
        this._element = _element;
        this._view = true;
        this.selectedChange = new EventEmitter();
        this.renderCalendar();
    }
    Object.defineProperty(Md2Clock.prototype, "selected", {
        get: function () { return this._selected; },
        set: function (value) {
            if (this._selected !== value) {
                this._selected = value || new Date();
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Md2Clock.prototype, "view", {
        set: function (value) {
            if (value === 'years') {
                this._view = false;
            }
            else {
                this._view = true;
            }
        },
        enumerable: true,
        configurable: true
    });
    /** Emits an event when the user selects a time. */
    Md2Clock.prototype._emitChangeEvent = function () {
        this.selectedChange.emit(this.selected);
    };
    /**
     * render Calendar
     */
    Md2Clock.prototype.renderCalendar = function () {
    };
    __decorate([
        Output(), 
        __metadata('design:type', EventEmitter)
    ], Md2Clock.prototype, "selectedChange", void 0);
    __decorate([
        Input(), 
        __metadata('design:type', Object)
    ], Md2Clock.prototype, "selected", null);
    __decorate([
        Input(), 
        __metadata('design:type', String), 
        __metadata('design:paramtypes', [String])
    ], Md2Clock.prototype, "view", null);
    Md2Clock = __decorate([
        Component({selector: 'md2-calendar',
            template: "",
            styles: [""],
            host: {
                'role': 'calendar'
            },
        }), 
        __metadata('design:paramtypes', [ElementRef])
    ], Md2Clock);
    return Md2Clock;
}());
